test/test_framework.py

```diff
@@ -117,3 +126,52 @@ def test_analysis_runtime_opts(self):
         target_opts["figure_names"] = None
 
         self.assertEqual(analysis.options.__dict__, target_opts)
+
+    def test_after_job_fail(self):
+        """Verify that analysis is cancelled in case of job failure"""
+
+        class MyExp(BaseExperiment):
+            """Some arbitraty experiment"""
+
+            def __init__(self, qubits):
+                super().__init__(qubits)
+                self.analysis = MyAnalysis()
+
+            def circuits(self):
+                circ = QuantumCircuit(1, 1)
+                circ.measure(0, 0)
+                return [circ]
+
+        class MyAnalysis(BaseAnalysis):
+            """Analysis that is supposed to be cancelled, because of job failure"""
+
+            def _run_analysis(self, experiment_data):
+                res = AnalysisResultData(name="should not run", value="blaaaaaaa")
+                return [res], []
+
+        class MyBackend(AerSimulator):
```

Comment:
Do you really need to run simulator? I think you can just use FakeBackend and override its run method to directly return MyJob instance.

Reply:
It's easy to do, but I'm curious - what would be the difference from the simulator?

Reply:
I don't really understand the purpose of this test. If this intends to return failed job, you don't need to run simulator and you can remove unnecessary dependency on Aer. If this intends to test some specific case for simulator, then it's worth having Aer simulator here.